torch/_inductor/utils.py

```diff
@@ -1291,17 +1291,10 @@ def _use_template_for_cpu(layout):
 
 
 def use_cpp_bmm_template(layout, mat1, mat2):
-    def stride_is_decreasing(x):
-        x.freeze_layout()
-        stride = x.get_stride()
-        return all(
-            not isinstance(stride[i], sympy.AtomicExpr) or stride[i] > stride[i + 1]
-            for i in range(len(stride) - 1)
-        )
-
-    return use_cpp_gemm_template(
-        layout, mat1, mat2, require_constant_mat2=False
-    ) and stride_is_decreasing(mat1)
+    return (
+        use_cpp_gemm_template(layout, mat1, mat2, require_constant_mat2=False)
+        and mat1.layout.is_contiguous()
```

Comment:
If stride is not contiguous, then selecting a single slice of the input X causes it to make an `ir.View` instead of `ir.ReinterpretView`, causing problems in the kernel member functions (view, slice_nd, index, etc.)